src/openapi/streaming/streaming.ts

```diff
@@ -398,6 +408,7 @@ class Streaming extends MicroEmitter<EmittedEvents> {
 
         switch (this.connectionState) {
             case this.CONNECTION_STATE_DISCONNECTED:
+                this.shouldSubscribeBeforeStreamingSetup = false;

```

Comment:
so the idea is we will keep the flag true initially so that all subscriptions can subscribe immediately when 'shouldSubscribeBeforeStreamingSetup' flag is true, and in case any disconnect/fail/reconnect happens we set this flag to false ,so then the flow works as it was previously

Reply:
yeap, thought that it would be enough to reset this flag just on `case this.CONNECTION_STATE_CONNECTED:`

Reply:
we don't want to try subscribe the 2nd time.

But maybe we can set it to false in one place earlier.. e.g. what about setting it to false, straight after whereever we use the value?

Reply:
e.g. line 944 ?

Reply:
the changes that i have currently done is on streaming level, so if i set the flag to false just after early subscribe of one subscription (i.e. line 944) this will then allow only that subscription to early subscribe , other subscription will not be able to subscribe immediately, since the `shouldSubscribeBeforeStreamingSetup` flag will be false now for later subscriptions

so i have made the changes considering the approach where we keep the flag true until we received any disconnect/fail/reconnect on the streaming so that initially all subscriptions can subscribe immediately.

Reply:
setting flag to false for `CONNECTION_STATE_CONNECTED` state as well as suggested